### PR TITLE
Updated pluralization in bravery panel (shields panel) to handle 0, 1, or more than 1

### PR DESCRIPTION
**Pull request by bsclifton**
- [x] Submitted a [ticket](https://github.com/brave/browser-laptop/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- [x] Ran `git rebase -i` to squash commits (if needed).

Updated pluralization in bravery panel (shields panel) to handle 0, 1, or more than 1

Also-
Hardened tests quite a bit for the Bravery Panel. This includes
- added new helpers for:
  - waitForTabCount
  - openBraveMenu
  - waitForDataFile
- added waitForTabCount in places where race condition was present
- increase timeout for the waitForDataFile portion, which sometimes took longer than 1 second
- added a logVerbose method to many helper methods. Can be toggled by setting variable "logVerboseEnabled" to true in `test/lib/brave.js`

**NOTE**: _tests are not included for these strings. I can either add them or create an issue to track this_

Fixes https://github.com/brave/browser-laptop/issues/4966

Auditors: @bbondy

Test Plan:
- Launch Brave, go to https://clifton.io, and open bravery panel
- Notice that:
  - "Ad or Tracker blocked" is used instead of the plural (since only 1 is present)
  - items with 0 use the plural tense
- Visit https://brianbondy.com and notice:
  - "Ads or Trackers blocked" is used since there are 3 present

### CODE REVIEW

**Comment by bbondy** [comment]
I think the l10n properties files have support for this with something like:

```
turtles = {[ plural(n) ]}
turtles[zero]  = The earth is not supported by turtles
turtles[one]   = The earth is supported by 1 turtle
turtles[other] = The earth is supported by {{n}} turtles
```

`data-l10n=id='turtles'`

**Comment by bbondy** [comment]
We have an in code example here:
findResultMatches

**Comment by bsclifton** [comment]
this is great! working on a fix now...

**Comment by bsclifton** [comment]
Updated and ready for re-review :smile: (sorry about the multiple rebases; wanted to make sure it's nice and clean)

File: test/lib/brave.js [hunk 1]
```diff
@@ -13,6 +13,14 @@ chai.use(chaiAsPromised)
 
 const Server = require('./server')
 
+// toggle me for more verbose logs! :)
+const logVerboseEnabled = false
```
Comment (bbondy) [inline test/lib/brave.js]: Please document this in the `docs/tests.md`

File: test/components/braveryPanelTest.js [hunk 2]
```diff
@@ -202,11 +187,14 @@ describe('Bravery Panel', function () {
...
-        .url(url)
-      yield openBraveMenu(this.app.client)
-      yield this.app.client
+        .ipcSend(messages.SHORTCUT_NEW_FRAME, url)
+        .waitForTabCount(2)
+        .waitForUrl(url)
+        .windowByUrl(Brave.browserWindowUrl)
+        // .tabByIndex(0)
+        // .loadUrl(url)
+        // .url(url)
```
Comment (bbondy) [inline test/components/braveryPanelTest.js]: Should these be removed?

**Comment by bbondy** [comment]
looks great, thanks.

**Pull request merged**